src/shared/compatibility.js

```diff
@@ -867,6 +867,11 @@ PDFJS.compatibilityChecked = true;
       base = new JURL(String(base));
     }
 
+    // Issue 8726: Safari needs url to be a String object.
+    // Otherwise, replace() below will produce a blank string.
+    if (isSafari) {
+      url = String(url);
```

Comment:
For reference: This part looks fairly similar to PR #8727, which was closed based on https://github.com/mozilla/pdf.js/pull/8727#issuecomment-328908500.
@yurydelendik What were the problems?

---
Also, please see https://github.com/mozilla/pdf.js/wiki/Squashing-Commits.